homeassistant/config.py

```diff
@@ -1249,8 +1247,24 @@ def extract_domain_configs(config: ConfigType, domain: str) -> Sequence[str]:
 
     Async friendly.
     """
-    pattern = re.compile(rf"^{domain}(| .+)$")
-    return [key for key in config if pattern.match(key)]
+    return [key for key in config if domain_from_config_key(key) == domain]
+
+
+def domain_from_config_key(key: str) -> str:
```

Comment:
This name is confusing as the return value can be a string that won't be a domain, eg `"hue  "`.

Reply:
OK, what do you suggest then? Should we raise in this case? Or just not return invalid domains?

Reply:
I'm not sure. On a high level I think we should call out invalid domains to the user and move on with the validation and setup of remaining valid domains. Invalid domains should be discarded as early as possible.

Reply:
Right, that makes sense 👍

Reply:
Do you still want the name of this function to be changed? We could state in the docstring that the passed key must be valid for it to return a valid domain?

Reply:
I'm afraid we'll use this incorrectly if we keep it around. Is there a use case after the config validation to have invalid domains be returned by this function?